core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -394,16 +410,22 @@ Builder specsById(Map<Integer, PartitionSpec> newSpecsById) {
     }
 
     Builder filterData(Expression newDataFilter) {
+      Preconditions.checkArgument(
+          deleteFiles == null, "Index constructed from files does not support data filters");
       this.dataFilter = Expressions.and(dataFilter, newDataFilter);
       return this;
     }
 
     Builder filterPartitions(Expression newPartitionFilter) {
+      Preconditions.checkArgument(
```

Comment:
Question, we can probably support partition filters easier than row filters, if i understand correctly?  (Even with this current code)

Reply:
Probably? I am not sure how useful that would be, though. I expect the filtering to happen outside, just like in the distributed planning case.